device-registry/src/main/scala/org/genivi/sota/device_registry/DevicesResource.scala

```diff
@@ -47,12 +47,14 @@ class DevicesResource(namespaceExtractor: Directive1[AuthedNamespaceScope],
 
   def searchDevice(ns: Namespace): Route =
     parameters(('regex.as[String Refined Regex].?,
-                'deviceId.as[String].?)) { // TODO: Use refined
-      case (Some(re), None) =>
-        complete(db.run(DeviceRepository.search(ns, re)))
-      case (None, Some(deviceId)) =>
+                'deviceId.as[String].?,
+                'offset.as[Long] ? 0,
+                'limit.as[Long] ? 50)) { // TODO: Use refined
+      case (Some(re), None, offset, limit) =>
+        complete(db.run(DeviceRepository.search(ns, re, offset, limit)))
+      case (None, Some(deviceId), _, _) =>
         complete(db.run(DeviceRepository.findByDeviceId(ns, DeviceId(deviceId))))
-      case (None, None) => complete(db.run(DeviceRepository.list(ns)))
+      case (None, None, offset, limit) => complete(db.run(DeviceRepository.list(ns, offset, limit)))
```

Comment:
I think we should return a different result here, something like:

```
{
  "values": [device0, device1...],
  "offset": 0,
  "total": 200
}
```

This way clients can just use those params to request different pages, instead of having to track what page they are on.


Ideally instead of count and total we would just give a resource identifier for next/previous (see HATEOS), but we are not there yet...